### PR TITLE
[rebase] OSSM-1423 Preliminary support for Gateway API

### PR DESCRIPTION
**Pull request by jewertow**
The most important part of this rebase are changes in [pilot/pkg/config/kube/gateway/deploymentcontroller.go](https://github.com/maistra/istio/pull/574/files#diff-c49c9e7fbae41236bcd0daff3ba0f8d2be118f9d4d33086ac91880ff653f7a4d) that I had to add to fix errors in istiod. Besides I changed a little bit role and cluster role that we apply in integration tests and adjusted tests of Gateway API to the current API of test framework.

### CODE REVIEW

File: tests/integration/servicemesh/maistra/testdata/role.yaml [hunk 1]
```diff
@@ -74,9 +64,6 @@ rules:
   - apiGroups: ["route.openshift.io"]
     resources: ["routes", "routes/custom-host"]
     verbs: ["get", "list", "watch", "create", "delete", "update"]
-  - apiGroups: ["maistra.io"]
-    resources: ["servicemeshextensions"]
-    verbs: ["get", "list", "watch"]
```
Comment (jewertow) [inline tests/integration/servicemesh/maistra/testdata/role.yaml:79]: These resources will no longer be needed.

File: tests/integration/servicemesh/maistra/testdata/role.yaml [hunk 2]
```diff
@@ -6,16 +6,6 @@ metadata:
...
-  # sidecar injection controller
-  - apiGroups: [ "admissionregistration.k8s.io" ]
-    resources: [ "mutatingwebhookconfigurations" ]
-    verbs: [ "get", "list", "watch", "update", "patch" ]
-
-  # configuration validation webhook controller
-  - apiGroups: [ "admissionregistration.k8s.io" ]
-    resources: [ "validatingwebhookconfigurations" ]
-    verbs: [ "get", "list", "watch", "update" ]
-
```
Comment (jewertow) [inline tests/integration/servicemesh/maistra/testdata/role.yaml:18]: I moved these resources to the cluster role, because they are cluster-scoped.

**Review by luksa — APPROVED** [review]

**Pull request merged**